src/pages-and-resources/xpert-unit-summary/settings-modal/SettingsModal.jsx

```diff
@@ -105,6 +110,82 @@ SettingsModalBase.defaultProps = {
   footer: null,
 };
 
+const ResetUnitsButton = ({
+  intl,
+  courseId,
+  checked,
+  visible,
+}) => {
+  const resetStatusRequestStatus = useSelector(getResetStatus);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (resetStatusRequestStatus === RequestStatus.SUCCESSFUL) {
+      setTimeout(() => {
+        dispatch(updateResetStatus({ status: '' }));
+      }, 2000);
+    }
+  }, [resetStatusRequestStatus]);
+
+  const handleResetUnits = () => {
+    dispatch(resetXpertSettings(courseId, { enabled: checked === 'true', reset: true }));
+  };
+
+  const getResetButtonState = () => {
+    switch (resetStatusRequestStatus) {
+    case RequestStatus.PENDING:
+      return 'pending';
+    case RequestStatus.SUCCESSFUL:
+      return 'finish';
+    default:
+      return 'default';
+    }
+  };
+
+  if (!visible) { return null; }
+
+  return (
+    <OverlayTrigger
+      placement="right"
+      overlay={(
+        <Tooltip id={`tooltip-reset-${checked}`}>
+          {intl.formatMessage(messages.resetAllUnitsTooltip, { state: checked === 'true' ? 'checked' : 'unchecked' })}
```

Comment:
Since `state` will be a replaced string on the translation, wouldn't it be better to have two different translation strings, one for each state?

Reply:
If you see the design, the text are the same, just one word is different

Reply:
But if you think about translations, both `checked` and `unchecked` words should also be able to be translated. In this case these words are being hardcoded and there's no way to translate them through the dictionary.

Reply:
Also, according to the Figma, the words should be `check` and `uncheck`, but even if you fix the words, the translation would not be possible.

Reply:
Fixed ✅